core/auth/auth.go

```diff
@@ -268,21 +282,36 @@ func getServiceAccountEmail(cfg *config.Configuration) string {
 }
 
 // getKey searches for a key in the following order: client configuration, environment variable, credentials file.
-func getKey(cfgKey, cfgKeyPath *string, envVar, credType credentialType, cfgCredFilePath string) error {
+func getKey(cfgKey, cfgKeyPath *string, envVarKeyPath, envVarKey, credTypePath, credTypeKey credentialType, cfgCredFilePath string) error {
```

Comment:
```suggestion
func getKey(cfgKey, cfgKeyPath *string, envVarKeyPath, envVarKey string, credTypePath, credTypeKey credentialType, cfgCredFilePath string) error {
```
`envVarKeyPath` and `envVarKey` aren't really of type `credentialType`, so they both can set as string. With that you also don't need to parse them into string in the calls `os.LookupEnv`